src/Parallel/ExtractFromInbox.hpp

```diff
@@ -0,0 +1,21 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+namespace Parallel {
```

Comment:
Please drop this commit, it doesn't aid readability and goes against our practice of "Use raw C++ as much as possible" These types of helper functions to save 20 characters in a few places just make the code more difficult to read, make it seem more magical than it is, and makes it so that people learn how to code in SpECTRE instead of C++. The last point is relevant for people who end up switching to industry and want to employ their C++ skills there. Furthermore, I can find endless resources on C++ but rather few on SpECTRE itself in comparison.

Reply:
I understand and agree with the idea to "Use raw C++ as much as possible". I still don't like the idea of copy-pasting this code in ~10 places in the linear solver code:

```cpp
  using inbox_tags = tmpl::list<InboxTag>;

  template <typename DbTags, typename... InboxTags, typename Metavariables,
            typename ArrayIndex>
  static bool is_ready(const db::DataBox<DbTags>& box,
                       const tuples::TaggedTuple<InboxTags...>& inboxes,
                       const Parallel::GlobalCache<Metavariables>& /*cache*/,
                       const ArrayIndex& /*array_index*/) noexcept {
    const auto& inbox = get<InboxTag>(inboxes);
    return inbox.find(db::get<TemporalIdTag>(box)) != inbox.end();
  }

  template <typename DbTagsList, typename... InboxTags, typename Metavariables,
            typename ArrayIndex, typename ActionList,
            typename ParallelComponent>
  static std::tuple<db::DataBox<DbTagsList>&&, bool, size_t> apply(
      db::DataBox<DbTagsList>& box, tuples::TaggedTuple<InboxTags...>& inboxes,
      Parallel::GlobalCache<Metavariables>& cache,
      const ArrayIndex& array_index, const ActionList /*meta*/,
      const ParallelComponent* const /*meta*/) noexcept {
    const auto received_data =
        std::move(tuples::get<InboxTag>(inboxes)
                      .extract(db::get<TemporalIdTag>(box))
                      .mapped());
  // ...
  }
```

I leaned the idea of using inheritance to implement common use cases from `Parallel/InboxInserters.hpp`.. that's also just standard C++ code.

Reply:
For the retrieval, it's only
```cpp
std::move(tuples::get<InboxTag>(inboxes)
                      .extract(db::get<TemporalIdTag>(box))
                      .mapped());
```
which is very minimal, easy to read, and I think actually much clearer than calling some wrapper.

For the `is_ready`, yes that's more code, but you're only currently removing it in 1 place (and doing so with actually changing the code), plus it makes the code much harder to read. With the change we have to again hunt down a struct rather than having the code self-contained.

If you want to do this to the elliptic code, that's fine, but it's not coming into the evolution code.

Reply:
For the retrieval I thought the `std::move` could easily be missed, which would add an unnecessary copy that probably wouldn't be caught by any tests, don't you think?

Reply:
> If you want to do this to the elliptic code, that's fine, but it's not coming into the evolution code.

I'd appreciate a less authoritative tone.. I value your opinion on these matters, but please don't make it sound like your word is final.

Reply:
Maybe, but I think it's better that people learn why they should use `std::move` there. Furthermore, people will look at existing code, so it's a matter of maintaining an existing standard (not trivial, I agree), and then new code will follow.